flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -82,6 +82,9 @@ public class ScanContext implements Serializable {
   private static final ConfigOption<Boolean> INCLUDE_COLUMN_STATS =
       ConfigOptions.key("include-column-stats").booleanType().defaultValue(false);
 
+  private static final ConfigOption<Integer> SNAPSHOT_GROUP_LIMIT =
+      ConfigOptions.key("snapshot-group-limit").intType().defaultValue(Integer.MAX_VALUE);
```

Comment:
I talked with @stevenzwu about this and the best name we could come up with was `max-planning-snapshot-count`. What do you think, @chenjunjiedada? Could you rename this?

Reply:
+1 to this name. Config keys should ideally be concise as well as being as short as possible and `max-planning-snapshot-count` achieves that.

The documentation can potentially use the language `Maximum number of snapshots to consume and plan per group in each iteration of an incremental scan` or something similar (might need to work on that language too but the language from the other ideas can be used in the docs possibly).

Reply:
Done.